axlearn/common/attention.py

```diff
@@ -1808,11 +1828,19 @@ def _compute_attention(
             k_proj: [batch_size, source_length, num_heads, per_head_dim].
             v_proj: [batch_size, source_length, num_heads, per_head_dim].
             attention_logit_biases: See ``On attention logit biases`` in the file comments.
+            segment_ids: See ``segment_ids`` in the file comments.
 
         Returns:
             The context of shape [batch_size, target_length, num_heads, per_head_dim],
             and probs of shape [batch, num_heads, target_length, source_length].
         """
+        if attention_logit_biases is not None and segment_ids is not None:
+            raise ValueError(
+                "Using both segment_ids and attention_logit_biases is not allowed. "
+                "If you have segment_ids, consider merging them into attention_logit_biases using "
+                "AttentionLogitBiasLayer."
+            )
```

Comment:
Would it be more accurate to say:
```suggestion
        if segment_ids is not None:
            raise ValueError(
                "segment_ids is not supported. To use segment_ids, construct attention_logit_biases using an "
                "AttentionLogitBiasLayer."
            )
```
since providing `segment_ids` without `attention_logit_biases` is going to result in silent behavior?